pkg/metrics/server/server.go

```diff
@@ -302,7 +305,9 @@ func (s *defaultServer) createListener(ctx context.Context, log logr.Logger) (ne
 		cfg.Certificates = []tls.Certificate{keyPair}
 	}
 
-	return tls.Listen("tcp", s.options.BindAddress, cfg)
+	l, err := s.options.ListenConfig.Listen(ctx, "tcp", s.options.BindAddress)
```

Comment:
Should we return the error here if listen already fails?

Not sure if we should call tls.NewListener if Listen failed

Reply:
@tomelliot16  ^^

Reply:
This was mentioned in this comment.  https://github.com/kubernetes-sigs/controller-runtime/pull/2519#discussion_r1340705060

/remove lgtm